mmaction/datasets/base.py

```diff
@@ -167,7 +167,7 @@ def evaluate(self,
             print_log(msg, logger=logger)
 
             if metric == 'top_k_accuracy':
-                topk = metric_dict.get('topk')
+                topk = metric_options['top_k_accuracy'].get('topk')
```

Comment:
what's the benefit of using `.get` if the default value is not provided?

Reply:
use `setdefault` for dict now.